src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlFormatReaderGenerator.cs

```diff
@@ -940,14 +932,7 @@ private void ThrowValidationException()
         static internal object UnsafeGetUninitializedObject(Type type)
         {
 #if !NET_NATIVE
-            if (type.GetTypeInfo().IsValueType)
-            {
-                  return Activator.CreateInstance(type);
-            }
-
-            const BindingFlags Flags = BindingFlags.Public | BindingFlags.Instance;
-            bool hasDefaultConstructor = s_typeHasDefaultConstructorMap.GetOrAdd(type, t => t.GetConstructor(Flags, Array.Empty<Type>()) != null);
-            return hasDefaultConstructor ? Activator.CreateInstance(type) : TryGetUninitializedObjectWithFormatterServices(type) ?? Activator.CreateInstance(type);
+            return FormatterServices.GetUninitializedObject(type) ?? Activator.CreateInstance(type);
```

Comment:
> the initial CoreFx implementation had this fallback

That's because the assembly and method were being loaded via reflection, and if they weren't found, then null would be returned and this fallback would be used.  That's not the case here, though.